packages/input-manager/src/integrations/xkeys/index.ts

```diff
@@ -0,0 +1,172 @@
+import { listAllConnectedPanels, setupXkeysPanel, XKeys } from 'xkeys'
+import { Logger } from '../../logger'
+import { Device } from '../../devices/device'
+import { Symbols } from '../../lib'
+import { ClassNames, SomeFeedback, Tally } from '../../feedback/feedback'
+
+enum Colors {
+	RED = '#ff0000',
+	GREEN = '#00ff00',
+	BLUE = '#0000ff',
+	WHITE = '#ffffff',
+	YELLOW = '#ffff00',
+	ORANGE = '#ff8000',
+}
+
+export interface XKeysDeviceConfig {
+	device: XKeysDeviceIdentifier
+}
+
+export interface XKeysDeviceIdentifier {
+	path?: string
+	serialNumber?: string
+	productId?: number
+	index?: number
+}
+
+export class XKeysDevice extends Device {
+	#config: XKeysDeviceConfig
+	#feedbacks: Record<number, SomeFeedback> = {}
+	#device: XKeys | undefined
+
+	constructor(config: XKeysDeviceConfig, logger: Logger) {
+		super(logger)
+		this.#config = config
+	}
+
+	async init(): Promise<void> {
+		const allDevices = listAllConnectedPanels()
+		const deviceInfo = allDevices.find((thisDevice, index) => {
+			let match = true
+			if (this.#config.device.path && thisDevice.path !== this.#config.device.path) match = false
+			if (this.#config.device.serialNumber && thisDevice.serialNumber !== this.#config.device.serialNumber)
+				match = false
+			if (this.#config.device.productId && thisDevice.productId !== this.#config.device.productId) match = false
+			if (this.#config.device.index && index !== this.#config.device.index) match = false
```

Comment:
I'm not sure if the returned order of the devices are guaranteed (this might be a future issue), but the `path` should point towards a specific usb-port so that should provide a unique match.

Secondly, we should use the xkey's `unitId` property to match the config device.
The `unitId` is a configurable setting in the xkey-panel, intended to be set by a user after the "labels have been glued on", ie the physical configuration has been done to the panel. This allows for having an identical backup-xkeys-panel with the same `unitId` and if the first one fails, the backup could be quickly plugged in and behave identically as the original.
  I would recommend changing this to:
```typescript
const allDevices = listAllConnectedPanels()
const useDevices = (
    await Promise.all(
        allDevices.map(async (thisDevice, index): Promise<XKeys | null> => {
            const config = this.#config.device
            if (config.productId && thisDevice.productId !== config.productId) return null

            const xkeysDevice = await setupXkeysPanel(thisDevice)

            // Try matching the unitId (unitId === 0 means that it hasn't been set)
            if (config.unitId !== 0 && xkeysDevice.unitId !== 0) {
                // We can use the userId to identify the xkeys-panel:
                if (config.unitId === xkeysDevice.unitId) return xkeysDevice
            } else {
                // No unitId available, try matching something else:

                // Note: The serialnumber isn't used in any existing xkeys today, but perhaps in the future?
                if (config.serialNumber && thisDevice.serialNumber) {
                    // We can use the serialnumber to match:
                    if (thisDevice.serialNumber === config.serialNumber) return xkeysDevice
                } else {
                    // Last try, matching the path:
                    if (config.path && thisDevice.path !== config.path) return xkeysDevice
                }
            }

            // nothing matched..
            await xkeysDevice.close()
            return null
        })
    )
).filter(Boolean) as XKeys[]

for (const device of useDevices) {
    // setup the device
}
```